prov/efa/src/efa_hmem.c

```diff
@@ -180,6 +230,21 @@ static int efa_hmem_support_status_update_synapseai(struct efa_hmem_support_stat
 
 	synapseai_status->initialized = true;
 	synapseai_status->p2p_supported = true;
+
+	/*  Only the long read protocol is supported */
+	if (-ENODATA != fi_param_get(&rxr_prov, "inter_max_medium_message_size", &tmp_value) ||
+		-ENODATA != fi_param_get(&rxr_prov, "inter_min_read_message_size", &tmp_value) ||
+		-ENODATA != fi_param_get(&rxr_prov, "runt_size", &tmp_value)) {
+		EFA_WARN(FI_LOG_DOMAIN,
+		         "One of the following environment variable(s) was set: (FI_EFA_INTER_MAX_MEDIUM_MESSAGE_SIZE, "
+		         "FI_EFA_INTER_MIN_READ_MESSAGE_SIZE, and/or FI_EFA_RUNT_SIZE), but EFA HMEM via Synapse API "
+		         "only supports long read protocol.  Please unset the env variable(s) and re-run.\n");
+		abort();
+	}
```

Comment:
Genuine question: Do these environment variables being set really warrant an `abort()`/failure? Could they not just be ignored after warning? I probably don't have enough context wrt the protocol selection.

Reply:
If customers request a feature but we cannot satisfy, we should abort().

Reply:
The warning will not be printed unless customers set `FI_LOG_LEVEL`. So silent ignorance will surprise our customers.

Reply:
We should abort immediately if the user requests a configuration which we cannot honor.  This way... if the run succeeds, the user always knows that the internal library used the user provided configurations.

Imagine being a user which wants to try  runtingread with various different message sizes.  You run once with runt_size == 1, and then again with runt_size == 100,000 and get the same result.  The user could then conclude that runt_size doesn't impact performance, but in reality we weren't using that protocol.  This change is to help avoid that situation.